contrib/aws/aws-sdk-go/aws/example_test.go

```diff
@@ -25,3 +31,33 @@ func Example() {
 		Bucket: aws.String("some-bucket-name"),
 	})
 }
+
+func TestWithErrorCheck(t *testing.T) {
+	testOpts := func(errExist bool, opts ...awstrace.Option) func(t *testing.T) {
+		return func(t *testing.T) {
+			mt := mocktracer.Start()
+			defer mt.Stop()
+
+			cfg := aws.NewConfig().WithRegion("us-west-2")
+			sess := session.Must(session.NewSession(cfg))
+			sess = awstrace.WrapSession(sess, opts...)
+
+			s3api := s3.New(sess)
+			s3api.CreateBucket(&s3.CreateBucketInput{
+				Bucket: aws.String("some-bucket-name"),
+			})
+
+			spans := mt.FinishedSpans()
+			assert.True(t, len(spans) > 0)
+
+			s := spans[len(spans)-1]
+			assert.Equal(t, errExist, s.Tag(ext.Error) != nil)
+		}
+	}
+
+	t.Run("defaults", testOpts(true))
+	t.Run("errcheck", testOpts(false, awstrace.WithErrorCheck(func(err error) bool {
+		return !strings.Contains(err.Error(), `InvalidAccessKeyId: The AWS Access Key Id you provided does not exist in our records`)
```

Comment:
The error that is actually received doesn't contain this text, so it will fail. For the test purposes, it's best to check for different content. `NoCredentialProviders: no valid providers in chain. Deprecated.` 
```suggestion
	t.Run("errcheck", testOpts(false, awstrace.WithErrorCheck(func(err error) bool {
		return !strings.Contains(err.Error(), `NoCredentialProviders: no valid providers in chain`)
```

Reply:
I ran the contrib test with [./test.sh --contrib](https://github.com/DataDog/dd-trace-go/blob/1f7d47aa87cdcfbfe5cbec0fb3773faddd3383c8/test.sh#L55) and every test under `contrib/aws/aws-sdk-go/aws` succeeded.